zerver/views/registration.py

```diff
@@ -523,6 +527,12 @@ def accounts_register(
             "sorted_realm_types": sorted(
                 Realm.ORG_TYPES.values(), key=lambda d: d["display_order"]
             ),
+            "default_email_address_visibility": None
+            if realm is None
+            else realm_user_default.email_address_visibility,
```

Comment:
This feels hard enough to read that I think we should compute a `default_email_address_visibility` above and just use its value here?